File: app/src/main/java/me/shikasd/twittershortcut/MainActivity.java
package me.shikasd.twittershortcut;

import android.app.Activity;
import android.content.ComponentName;
import android.content.Intent;
import android.os.Bundle;

public class MainActivity extends Activity {
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Intent intent = new Intent();
        intent.addCategory("android.intent.category.LAUNCHER");
        intent.setAction("android.intent.action.MAIN");
        intent.setComponent(new ComponentName("com.twitter.android", "com.twitter.android.StartActivity"));
        startActivity(intent);

        finish();
    }
}

File: app/src/main/kotlin/me/shikasd/twittershortcut/MainActivity.kt
package me.shikasd.twittershortcut

import android.app.Activity
import android.content.ComponentName
import android.content.Intent
import android.os.Bundle


class MainActivity : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val intent = Intent()
        intent.addCategory("android.intent.category.LAUNCHER")
        intent.action = "android.intent.action.MAIN"
        intent.component = ComponentName("com.twitter.android", "com.twitter.android.StartActivity")
        startActivity(intent)

        finish()
    }
}